jobs/build/tarball-sources/Jenkinsfile

```diff
@@ -0,0 +1,141 @@
+                        defaultValue: [
+                            'release-engineering@redhat.com',
+                        ].join(',')
+                    ],
+                    [
+                        name: 'MAIL_LIST_FAILURE',
+                        description: 'Failure Mailing List',
+                        $class: 'hudson.model.StringParameterDefinition',
+                        defaultValue: [
+                            'aos-art-automation+failed-release@redhat.com'
+                        ].join(',')
+                    ],
+                    commonlib.mockParam(),
+                ]
+            ],
+        ]
+    )
+
+    commonlib.checkMock()
+    buildlib.initialize()
+    buildlib.cleanWorkdir("${env.WORKSPACE}")
+
+    try {
+        sshagent(['openshift-bot']) {
+            advisory = params.ADVISORY ? Integer.parseInt(params.ADVISORY.toString()) : 0
+            def components = params.COMPONENTS.replaceAll(',', ' ').split(' ')
+            def filename = "tarball-srouce-output.txt"
+            def outdir = params.OUT_DIR.toString()
+            def rcm_guest = params.RCM_GUEST.toString()
+            currentBuild.description += "tarball: ${components}"
+
+            stage("elliott tarball-source") {
+                component_args = ""
+                components.each {
+                    component_args += "--component ${it}"
+                }
+                buildlib.elliott """
+                    tarball-sources create
+                    --out-dir=${outdir}
+                    --force
+                    ${component_args}
+                    ${advisory} > ${filename}
+                """
+            }
+
+            stage("rsync") {
+                cmd = "rsync -avz --chmod=go+rX ${outdir} ${rcm_guest}"
+                commonlib.shell(
+                    script: cmd,
+                    returnAll: true
+                )
+            }
+
+            stage("file ticket to RCM") {
+                cmd = """awk '/${advisory}/ { save=\$0 }END{ print save }' ${filename}"""
+                verify = commonlib.shell(
+                        script: cmd,
+                        returnStdout: true
+                )
+
+                description = """
+                    Hi RCM,
+                    The OpenShift ART team needs to provide sources for `${components}` in https://errata.devel.redhat.com/advisory/${advisory}
+                    This release is urgent and we want it to be unblocked and shipped ASAP.
+
+                    The following sources are uploaded to rcm-guest.app.eng.bos.redhat.com:/mnt/redhat/staging-cds/
+
+                    ${verify}
+
+                    Attaching source tarballs to be published on ftp.redhat.com as in https://projects.engineering.redhat.com/browse/RCMTEMPL-6549
+                    """
+                
+                cmd = """jira create --noedit -p RCM -i Task -o summary='OCP Tarball sources' -o description='${description}' """
```

Comment:
does this actually work with a multiline description?

i'm a little worried anyway because someone could edit the description and introduce a `'` not realizing that it will break this. can the description be written to a file first and just referrred to?

Reply:
```
[jenkins@buildvm ~]$ jira create -h
usage: jira create [<flags>]

Create issue

Global flags:
      --help                   Show context-sensitive help (also try --help-long and --help-man).
  -v, --verbose ...            Increase verbosity for debugging
  -e, --endpoint=ENDPOINT      Base URI to use for Jira
  -k, --insecure               Disable TLS certificate verification
  -Q, --quiet                  Suppress output to console
      --unixproxy=UNIXPROXY    Path for a unix-socket proxy
      --socksproxy=SOCKSPROXY  Address for a socks proxy
  -u, --user=USER              user name used within the Jira service
      --login=LOGIN            login name that corresponds to the user used for authentication

Optional flags:
  -b, --browse                 Open issue(s) in browser after operation
      --editor=EDITOR          Editor to use
  -t, --template=TEMPLATE      Template to use for output
      --noedit                 Disable opening the editor
  -p, --project=PROJECT        project to create issue in
  -i, --issuetype=ISSUETYPE    issuetype in to create
  -m, --comment=COMMENT        Comment message for issue
  -o, --override=OVERRIDE ...  Set issue property
      --saveFile=SAVEFILE      Write issue as yaml to file
```
so far I didn't find any options looks like it can read from a file directly, I think it has the ability to construct commands from the config file, but that's a little bit complicated since that config file need to store on buildvm

Reply:
I am trying the approach pointed here https://github.com/go-jira/jira/issues/136, there's some problem with finding the template `ERROR Invalid Usage: No Template found for "./create"`

Reply:
another alternative way is we use shell env instead of para `${description}`